### DISCUSSION TITLE
Update Awareness State on Server side?

### DISCUSSION THREAD

**Discussion by LilithWittmann** [ueberdosis/hocuspocus#454] [Q&A]
Hey everyone,
Is there currently a way to update the awareness state on the server side?

My current issue is that I would like to update the user/color for the collaboration cursor on the backend side after the user authentication.

Thanks!

**Comment by canadaduane** [comment]
There is a now a [DirectConnection](https://github.com/ueberdosis/hocuspocus/blob/main/packages/server/src/DirectConnection.ts) class on the server that should let you get a handle on the document you want, and then the Awareness object (document.awareness). I haven't tried setting the awareness from the server after that, but I think it should be possible?
